eng/tools/dependency-testing/templates/package.json

```diff
@@ -9,7 +9,7 @@
   "scripts": {
     "build": "tsc -p .",
     "integration-test:browser": "karma start --single-run",
-    "integration-test:node": "mocha -r esm-workaround.js -r esm --require source-map-support/register --reporter mocha-multi-reporter.js  --reporter-option output=test-results.xml --timeout 350000 --full-trace \"dist-esm/**/{,!(browser)/**/}*.spec.js\" --exit",
+    "integration-test:node": "mocha --require tsx --require source-map-support/register --reporter mocha-multi-reporter.js  --reporter-option output=test-results.xml --timeout 350000 --full-trace \"dist-esm/**/{,!(browser)/**/}*.spec.js\" --exit",
```

Comment:
Should esm-workaround.js be removed as part of this update?

Reply:
Good Q, it's used for min/max testing and (as of now) for package integration testing. We're working on removing all of this but it'll be a few steps of work:

1. Merge this PR
2. Merge https://github.com/Azure/azure-sdk-for-js/pull/28826
3. Let things bake and ensure nightly builds are happy, min/max is happy, etc
4. Remove esm-workaround.js
5. Remove `esm` as a dependency from all our packages in a NO_CI commit

I think that would cover it